extensions/interactions/rules.json

```diff
@@ -0,0 +1,186 @@
+{
+  "CodeRepl": {
```

Comment:
I actually think I lean toward rules belonging as part of their definition of the interaction used by them. Although it's convenient to put it in a rules.json file, we're sort of in this awkward midway between having rules separate from interactions and having them part of the definition. From what I understand, we'd rather have interactions be self-contained (especially since rules and objects are the only parts of an interaction not defined directly within it). I don't think we should skimp on this particular effort, if it's not too unreasonable to put them within the respective interaction definitions.

Unless I'm mistaken, we really only need rule descriptions for validating explorations. The only other locations it exists (backend classification and backend rule tests) are/can both being/be moved to the frontend. I think this leads to us to requiring roughly one semi-complicated function for retrieving these descriptions in an effort to validate explorations. I think that's worthwhile.